src/data-types/mailstream.c

```diff
@@ -246,12 +218,13 @@ ssize_t mailstream_read(mailstream * s, void * buf, size_t count)
   }
 
   if (left > s->buffer_max_size) {
-    read_bytes = read_through_buffer(s, cur_buf, left);
+    read_bytes = mailstream_low_read(s->low, cur_buf, left);
```

Comment:
The other usage of mailstream_read() is here: mailstream_read_line_append() but it seems that it's going to be alright here too.